tests/test_black.py

```diff
@@ -1680,16 +1681,18 @@ def test_reformat_one_with_stdin_and_existing_path(self) -> None:
             report.done.assert_called_with(expected, black.Changed.YES)
 
     def test_reformat_one_with_stdin_empty(self) -> None:
-        with patch("sys.stdout", open(os.devnull, "w")):
-            # Patch `sys.stdout` to null device lest
-            # `format_stdin_to_stdout` detaches stdout at the end and causes
-            # "ValueError: I/O operation on closed file" in subsequent tests
-            black.format_stdin_to_stdout(
-                fast=True,
-                content="",
-                write_back=black.WriteBack.YES,
-                mode=DEFAULT_MODE,
-            )
+        output = io.StringIO()
+        with patch("io.TextIOWrapper", lambda *args, **kwargs: output):
+            try:
+                black.format_stdin_to_stdout(
+                    fast=True,
+                    content="",
+                    write_back=black.WriteBack.YES,
+                    mode=DEFAULT_MODE,
+                )
+                assert output.getvalue() == ""
```

Comment:
This line should be outside the try-except to ensure it's executed.

Reply:
Right, thanks. It's fixed!